src/sentry/search/events/filter.py

```diff
@@ -851,6 +852,50 @@ def _environment_filter_converter(
         else:
             return env_conditions[0]
 
+    def _project_slug_filter_converter(
+        self,
+        search_filter: SearchFilter,
+        _: str,
+    ) -> Optional[WhereType]:
+        """Convert project slugs to ids and create a filter based on those
+
+        This is cause we only store project ids in clickhouse
+        """
+        value = search_filter.value.value
+
+        if search_filter.operator == "=" and value == "":
+            raise InvalidSearchQuery(
+                'Cannot query for has:project or project:"" as every event will have a project'
+            )
+
+        slugs = to_list(value)
+        projects = {
+            p.slug: p.id
+            for p in Project.objects.filter(
+                id__in=self.params.get("project_id", []), slug__in=slugs
+            )
+        }
+        missing: List[str] = [slug for slug in slugs if slug not in projects]
+        if missing and search_filter.operator in EQUALITY_OPERATORS:
+            raise InvalidSearchQuery(
+                f"Invalid query. Project(s) {', '.join(missing)} do not exist or are not actively selected."
+            )
+        # Sorted for consistent query results
+        project_ids = list(sorted(projects.values()))
+        if project_ids:
+            # Create a new search filter with the correct values
+            converted_filter = self.convert_search_filter_to_condition(
+                SearchFilter(
+                    SearchKey("project.id"),
+                    search_filter.operator,
+                    SearchValue(project_ids if search_filter.is_in_filter else project_ids[0]),
+                )
+            )
+            if converted_filter:
+                if search_filter.operator in EQUALITY_OPERATORS:
+                    self.projects_to_filter.update(project_ids)
```

Comment:
new way of doing the `project_to_filter` method, afaik we don't do INEQUALITY here, since more complicated boolean searches can result in a project being added back in.

Reply:
This will convert something like `project:a project:b` into an `OR` rather than an `AND` condition right?

Reply:
Nah so the actual conditions will still be correct (see L887 the converted_filter will add `WHERE project = 1` correctly for example) this `projects_to_filter` list is cause we need to send a root(base?) condition on project_id
eg.
params[project_id] = [1,2,3]
condition = (project:1 and transaction:foo) or (project:2 and transaction:bar)

results in
select...where project_ids IN [1,2] and ((project=1 and txn=foo) or (project=2 and txn=bar))


Though there's currently a bug with this that's in the backlog(ie. i don't want to fix as part of the migration) something like this:
eg.
params[project_id] = [1,2,3]
condition = (project:1 and transaction:foo) or (transaction:bar)

results in
select...where project_ids IN [1] and ((project=1 and txn=foo) or (txn=bar))